### DISCUSSION TITLE
Different color icons

### DISCUSSION THREAD

**Discussion by Aman9das** [Arcticons-Team/Arcticons#342] [Ideas]
Continuation of #37
Basically having the icons in colors other than black, white.

**Comment by Donnnno** [comment]
I'm not feeling it, besides black & white personally. Mostly because the idea of Arcticons is to provide a distraction-free experience for your home screen.

And if there were to come different colors something like [Lines Chroma](https://play.google.com/store/apps/details?id=com.natewren.lineschroma) would be the way to go.

**Reply by Aman9das** [reply]
I was thinking something like community maintained variants..

Say i want a sky blue icon pack.. How great would it be to just go to the Arcticons page

- find a "Color_Variants_Catalog.md" and get the apk built by someone with a similar favourite colour.

- find a "Color_Variants_Creation_Guide.md" and build it myself and brag "i built an app"

- be lazy and request about it

**Reply by Donnnno** [reply]
Sure! We can do something like that! 

> "i built an app"

Haha, mood.

**Reply by Aman9das** [reply]
Obviously I'll build scripts and write guides to streamline the process

It won't cause you any headache this way, running raster.sh is pretty heavy afterall.

**Reply by Donnnno** [reply]
Yeah, maintaining Dark & Light is enough for me at the moment, mainly because of work, school & internship haha. 
Running the raster.sh for those two flavors isn't a big deal because I just can do that on my pc or laptop while working on other stuff at the same time.

**Comment by Aman9das** [comment]
So after the migration the solution would seem to be building custom apk like so:

- Fork Arcticons.

- create branch "color_name"

- clone locally

- replace #000 in black.sh with #color_code

- move all svgs from /icons/black to /other

- run "sh other/raster.sh" in bash

- build ArcticonsLight

And get the custom apk

**Reply by Donnnno** [reply]
yup

**Reply by Aman9das** [reply]
Ok so this should be on the next roadmap..

and create a new discussion on roadmap suggestions maybe?

**Reply by mubashir-rehman** [reply]
The community member should just create a PR with color hex and color name in a file say requestColor.txt or request.txt with hex on first line and name on second. if only hex is provided then get color name from https://github.com/meodai/color-names using github actions.

**Comment by Aman9das** [comment]
Ok so I'll write the beginning of Color_Variants_Guide

And I'll begin the Catalog too

But we need a roadmap / vision a lot.. The icon pack has a lot of potential:

- It'll be a bit of work but making a proper coloured version with the help of the community, just recoloring the lines..

- A duotone version?

- A filled flavour (coloured) different from frost

**Reply by mubashir-rehman** [reply]
Currently as per rules, we should stick to only one color of lines in the whole icon pack. And new community builds should be named after the requester's provided color name or color name got from https://github.com/meodai/color-names using github actions. Also maintain a file containing all color names already used so that there aren't any clashes(which can be checked by github action too)

**Comment by mubashir-rehman** [comment]
@Aman9das Can you please have a look at discussion on #403 . I am sorry for being late.

**Reply by Aman9das** [reply]
Yeah the option 2 is the more plausible one, and we're working on it, hopefully making custom flavors gets streamlined and easy by next week

But it'll have to be community maintained, too much work for 1 person..

**Reply by mubashir-rehman** [reply]
What do you mean by community maintained? I think github-actions is being used for now in nikgapps custom configs. Anyway, thanks a lot.

**Reply by mubashir-rehman** [reply]
#410 can also be used as a reference for #37
Also, the following link might also give some help for current discussion and #37 
https://github.com/nikgapps/deploy/blob/main/.github/workflows/Config_Deployment.yml
